ui/lib/CreateStream.tsx

```diff
@@ -19,38 +21,127 @@ export interface CreateStreamProps {
   // TODO: Support dynamically fetching list of accepted tokens in sdk::StreamManager
   tokenAddress: Address;
   amountPerSecond: number;
-  reasonCode: string;
   cart?: ReactNode;
   registerStream: (stream: Stream) => void;
+  renderReasonCode: () => Promise<string>;
+  handleTransactionStatus: (processing: boolean, processed: boolean, error: Error | null) => void;
 }
 
 const CreateStream = (props: CreateStreamProps) => {
+  const [nativeBalance, setNativeBalance] = useState<number | null>(null);
+  const [tokenBalance, setTokenBalance] = useState<number | null>(null);
+  const [gasPrice, setGasPrice] = useState<number | null>(null);
+  const { data: feeData } = useFeeData();
   const { address } = useAccount();
+
+  // Get balances for native tokens (or set to 1 after 5 seconds and keep fetching)
+  useEffect(() => {
+    let balanceCountdown: NodeJS.Timeout;
+    let balanceCountdownTriggered = false;
+
+    if (address) {
```

Comment:
add a comment on why this is used to help the user progress through the sequence in scenarios where there is some error